missioncontrol/etl/measure.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+import logging
+from distutils.version import LooseVersion
+
+from django.core.cache import cache
+
+from missioncontrol.celery import celery
+from missioncontrol.settings import DATA_EXPIRY_INTERVAL
+from .presto import raw_query
+from .schema import (CHANNELS, TELEMETRY_PLATFORM_MAPPING, get_measure_cache_key)
+from .versions import get_firefox_versions
+
+logger = logging.getLogger(__name__)
+
+
+@celery.task
+def update_measure(platform_name, channel_name, measure_name):
+    logger.info('Updating measure: %s %s %s', channel_name, platform_name,
+                measure_name)
+    channel = CHANNELS[channel_name]
+    cache_key = get_measure_cache_key(platform_name, channel_name, measure_name)
+
+    # data is stored in this format:
+    # {
+    #   'buildid': {
+    #     'version': 'XXX',
+    #     'data': [[<timestamp>, value, usage hours], [<timestamp>, value, usage hours], ...]
+    #   },
+    #   'buildid2': {
+    #       ...
+    #   },
+    #  ...
+    # }
+    min_timestamp = datetime.datetime.utcnow() - DATA_EXPIRY_INTERVAL
+    min_buildid_timestamp = min_timestamp - channel['update_interval']
+    data = cache.get(cache_key, {})
+    if data:
+        min_timestamp = max([min_timestamp] +
+                            [existing_build_data['data'][-1][0]
+                             for existing_build_data in data.values()])
+
+    # also place a restriction on version (to avoid fetching data
+    # for bogus versions)
+    versions = get_firefox_versions()
+    current_version = versions[channel_name]
+    if channel == 'esr':
+        min_version = str(LooseVersion(versions[channel_name]).version[0] - 7)
+    else:
+        min_version = str(LooseVersion(versions[channel_name]).version[0] - 1)
+
+    raw_sql = '''
+        select window_start, build_id, version, sum({}) as {}, sum(usage_hours) as usage_hours
+        from error_aggregates where application=\'Firefox\' and
+        version >= \'{}\' and version <= \'{}\' and
+        build_id > \'{}\' and
+        os_name=\'{}\' and
+        channel=\'{}\' and
+        window_start > timestamp \'{}\'
+        group by (window_start, build_id, version)
+        '''.format(measure_name, measure_name, min_version, current_version,
+                   min_buildid_timestamp.strftime('%Y%m%d'),
+                   TELEMETRY_PLATFORM_MAPPING[platform_name],
+                   channel_name,
+                   min_timestamp.strftime("%Y-%m-%d %H:%M:%S")).replace('\n', '').strip()
+    logger.info('Querying: %s', raw_sql)
+
+    for (window_start, build_id, version, measure_count, usage_hours) in raw_query(raw_sql):
+        # skip datapoints with no measures / usage hours
+        if not measure_count or not usage_hours:
+            continue
+
+        # the minimum build id for the query is based on minimum
```

Comment:
This `update_measure` task has no docstrings so it would be nice if this comment and the one containing a description of the schema were part of the docstrings, e.g.
```python
@celery.task
def update_measure(platform_name, channel_name, measure_name):
    """"Fetch latest data from error_aggregates and store it in redis

    The data is stored in the following format:
    data is stored in this format:
    {
        'buildid': {
          'version': 'XXX',
          'data': [[<timestamp>, value, usage hours], [<timestamp>, value, usage hours], ...]
        },
        'buildid2': {
          ...
        },
     ...
    }

    Every channel has a different minimum build_id/timestamp,
    defined in the settings. For any given point later than that the
    minimum timestamp we may reject it because it's too old.
    """"